http-webhook.md

```diff
@@ -0,0 +1,376 @@
+# HTTP 1.1 Web Hooks for Event Delivery
+
+## Abstract
+
+"Webhooks" are a popular pattern to deliver notifications between applications
+and via HTTP endpoints. In spite of pattern usage being widespread, there is no
+formal definition for Web Hooks. This specification aims to provide such a
+definition for use with [CNCF CloudEvents][CE], but is considered generally
+usable beyond the scope of CloudEvents.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to HTTP](#12-relation-to-http)
+2. [Delivering notifications](#2-delivering-notifications)
+3. [Authorization](#3-authorization)
+4. [Abuse Protection](#4-abuse-protection)
+5. [References](#5-references)
+
+## 1. Introduction
+
+["Webhooks"][Webhooks] are a popular pattern to deliver notifications between
+applications and via HTTP endpoints. Applications that make notifications
+available, allow for other applications to register an HTTP endpoint to which
+notifications are delivered.
+
+This specification defines a HTTP method by how notifications are delivered by
```

Comment:
Delivery/receiving callback requests is just part of the Web hook mechanism. As web hook was originally covered in @progrium's [blog](http://progrium.com/blog/2007/05/03/web-hooks-to-revolutionize-the-web/)
> To support web hooks, you allow the user to specify a URL where your application will post to and on what events.

So far actual [hook](https://stackoverflow.com/a/467568/381140) API part and related protocol like:
* API how callbacks are to be registered
** how should callback URL attribute be named
** how one defines for which events should callback URL be called (potential CloudEvents attribute usage?)
** negotiating content type for web hook callback
** verifying callback URL is valid and owned by the calling party
** have can all of this information be exchanged (query params, request body, via some 3rd party storage/API like k8s API CRD, ...)
* how can receiving party communicate it's no longer interested in the events
* how can delivery party share and interested party discover for which events hook can be registered, etc.
are not covered, yet.

Would it make sense to explicitly mention that some aspects of Web hook mechanism for reasons have been deliberately left out for now?

Reply:
I am indeed very deliberately leaving quite a few aspects mentioned above, except for the abuse protection handshake, because I don't think we can standardize them at this point.

For instance, in Azure, you are registering webhooks into Event Grid through the Azure Resource Manager API, as a subscription on a topic supplying filters, and that's a single interaction in a large framework of system-wide conventions applying to thousands of APIs. 

It's presently not realistic to expect for that interaction to be superseded by an API standard that this WG comes up with. First, because of the greater platform convention context. Second, because we're far from ready to commit to constraints for such an API.